core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -627,7 +627,16 @@ private[spark] class BlockManager(
   override def getLocalBlockData(blockId: BlockId): ManagedBuffer = {
     if (blockId.isShuffle) {
       logDebug(s"Getting local shuffle block ${blockId}")
-      shuffleManager.shuffleBlockResolver.getBlockData(blockId)
+      try {
+        shuffleManager.shuffleBlockResolver.getBlockData(blockId)
+      } catch {
+        case e: IOException =>
```

Comment:
Nit/ question: Could we move the if up as part of the case and avoid the need for explicit rethrow?

Reply:
It's because we should access the normal access path by default. We are able to use `Fallback` path only if the normal access path fail because `conf.get(config.STORAGE_DECOMMISSION_FALLBACK_STORAGE_PATH).isDefined` doesn't mean the fallback storage has the data.

Reply:
Please let me know if I misunderstand your comment.

Reply:
Yeah so I mean we still have a try/catch just the case statement has the if as part of it (eg “ Using if expressions in case statements”)